client/allocrunner/taskrunner/logmon_hook.go

```diff
@@ -69,28 +71,58 @@ func (h *logmonHook) launchLogMon() error {
 	return nil
 }
 
+func reattachConfigFromHookData(data map[string]string) (*plugin.ReattachConfig, error) {
+	if data == nil || data["reattach_config"] == "" {
+		return nil, nil
+	}
+
+	var cfg *shared.ReattachConfig
+	err := json.Unmarshal([]byte(data["reattach_config"]), cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return shared.ReattachConfigToGoPlugin(cfg)
+}
+
 func (h *logmonHook) Prestart(ctx context.Context,
 	req *interfaces.TaskPrestartRequest, resp *interfaces.TaskPrestartResponse) error {
 
-	// Launch logmon instance for the task.
-	if err := h.launchLogMon(); err != nil {
+	reattachConfig, err := reattachConfigFromHookData(req.HookData)
+	if err != nil {
+		h.logger.Error("failed to load reattach config", "error", err)
+		return err
+	}
+
+	// Launch or reattach logmon instance for the task.
+	if err := h.launchLogMon(reattachConfig); err != nil {
 		h.logger.Error("failed to launch logmon process", "error", err)
 		return err
 	}
 
-	err := h.logmon.Start(&logmon.LogConfig{
-		LogDir:        h.config.logDir,
-		StdoutLogFile: fmt.Sprintf("%s.stdout", req.Task.Name),
-		StderrLogFile: fmt.Sprintf("%s.stderr", req.Task.Name),
-		StdoutFifo:    h.config.stdoutFifo,
-		StderrFifo:    h.config.stderrFifo,
-		MaxFiles:      req.Task.LogConfig.MaxFiles,
-		MaxFileSizeMB: req.Task.LogConfig.MaxFileSizeMB,
-	})
+	if reattachConfig == nil {
```

Comment:
I think we should make `logmon.Start` idempotent and always call it. It seems _really_ unlikely but we could get in a situation where we start logmon but crash before `Start` is called.

Reply:
If that happened, we'd not have saved the reattach config, so it would still get a clean start?

Reply:
True, which is kind of gross because it means we'll have leaked a logmon process. I guess lets not worry about that for now since its such a rare edge case.